### PR TITLE
Delete istio-iptables.sh and istio-clean-iptables.sh

### PR DESCRIPTION
**Pull request by abhide**

### CODE REVIEW

**Comment by abhide** [comment]
FYI: PR: https://github.com/istio/istio/pull/18962 started making use of golang implementation

File: tests/scripts/istio-iptables-test.sh [hunk 1]
```diff
@@ -57,15 +57,9 @@ function compareWithGolden() {
   local FILE_UNDER_TEST
 
   case "${TEST_MODE}" in
-   "script")
-    FILE_UNDER_TEST="${SCRIPT_DIR}/../../tools/packaging/common/istio-iptables.sh"
-   ;;
    "golang")
```
Comment (howardjohn) [inline tests/scripts/istio-iptables-test.sh]: Should we just make this a proper golang test? It probably should have been this way from the start and just shelled out to the shell script, unless there is some reason it should be like this? Could be a follow up pr

**Comment by howardjohn** [comment]
Looks like the sidecar.deb is stil using the old shell script

File: tests/scripts/istio-iptables-test.sh (hunk 1 above)
Comment (abhide) [reply tests/scripts/istio-iptables-test.sh]: yeah. we can fix this in a separate PR

**Comment by abhide** [comment]
> Looks like the sidecar.deb is stil using the old shell script

@howardjohn Not quite sure where this is failing?
I don't see where its picking up `tools/packaging/common/istio-iptables` from?

```make: *** No rule to make target 'tools/packaging/common/istio-iptables', needed by '/tmp/tmp.1TYxJtF8Qd/build/work/out/linux_amd64/release/istio-sidecar.deb'.  Stop.```

**Comment by howardjohn** [comment]
It should be from the sidecar.deb make target I think? I can look closer
later

**Comment by rlenglet** [comment]
`tools/packaging/deb/istio.mk` currently contains:
```
ISTIO_DEB_DEST:=... \
                ${ISTIO_DEB_BIN}/istio-iptables \
                ...
$(foreach DEST,$(ISTIO_DEB_DEST),\
        $(eval ${ISTIO_OUT}/istio-sidecar.deb:   tools/packaging/common/$(notdir $(DEST))) \
        ...
```
So `make` will evaluate the following rule:
```
${ISTIO_OUT}/istio-sidecar.deb: tools/packaging/common/istio-iptables
```
`tools/packaging/common/istio-iptables` does not exist (unlike `tools/packaging/common/istio-iptables.sh`), which is why `make` fails.

You need to remove the `istio-iptables` and `istio-clean-iptables` paths from the `ISTIO_DEB_DEST` variable.

Instead, I'd suggest trying adding `istio-iptables istio-clean-iptables` to the `SIDECAR_DEB_DEPS`, which already contains the other Golang binary dependencies.

**Comment by abhide** [comment]
Not sure where this fails. 
`make istio-demo.yaml` works for me. 

Did a `git grep Copying` in repo. Don't see any place where `istio-iptables.sh` is getting copied.

@rlenglet @howardjohn any ideas?

**Comment by howardjohn** [comment]
https://github.com/istio/release-builder/blob/ce17e987ee032f07ae86b06750ee5c3e071b9a46/pkg/build/archive.go#L52

**Comment by abhide** [comment]
Thanks @howardjohn. I guess I need to remove it from that repo first then.
I guess we branch all repos identically? Just want to make sure remove that from `release-builder` repo `master` branch won't impact any maintenance/patch release for previous versions?

**Comment by howardjohn** [comment]
Yes we branch (almost) every repo the same. Changes to master won't impact
old releases. Furthermore, the SHA of the release-builder we use is pinned,
so you can make the change and it won't impact other PRs that still depend
on that file existing, since it won't be updated until your PR merges,
which will update the SHA as well

**Comment by abhide** [comment]
Thanks for confirming @howardjohn. Pushed a PR to `istio/release-builder`

**Comment by abhide** [comment]
Thanks @howardjohn. update SHA in `prow/release-commit.sh`

**Comment by howardjohn** [comment]
lgtm but we should wait for a resolution to https://github.com/istio/istio/issues/19380 before merging?

**Comment by lambdai** [comment]
#19380 is closed
@howardjohn Do you have other concerns?

**Pull request merged**